pyiceberg/table/update/spec.py

```diff
@@ -280,7 +280,7 @@ def _partition_field(self, transform_key: Tuple[int, Transform[Any, Any]], name:
             for field_key in historical_fields:
                 if field_key[0] == source_id and field_key[2] == repr(transform):
                     if name is None or field_key[3] == name:
-                        return PartitionField(source_id, field_key[1], transform, name)
+                        return PartitionField(source_id, field_key[1], transform, field_key[3])
```

Comment:
What do you think about using an Enum to get the values from the field_key? I think that will make it easier to understand what each position means.